### DISCUSSION TITLE
UEFI modding on Razer laptop

### DISCUSSION THREAD

**Discussion by terminatorul** [terminatorul/NvStrapsReBar#66] [General]
@stefanstgs  https://github.com/terminatorul/NvStrapsReBar/issues/1#issuecomment-2004127107:

>Hi!
>I have a guide on how to fully open bios on razer models up to 10875h or maybe even later. I think it also applies to other models than razer.
>
>My question is how exactly are we supposed to modify bios file if on my laptop both GPU and UEFI are both on the same chip (i think)
>
>Let me know if the guide I mentioned here is usefull for anyone, I could post it here

Yes, please post (or link) your guide here.

Can you give some system information, so I can understand your question with GPU and UEFI on the same chip ?

For discrete GPUs, the card does have some UEFI modules on it, but that is called vBIOS and is signed (and encrypted) directly by NVIDIA, so users and UEFI modders can not tinker with it.

For integrated GPUs, UEFI is actually on a dedicated BIOS chip on the motherboard, so not the same as the GPU ...

And then there are some sistems that have an iGPU close to the level of a discrete GPU: systems like Apple M2 chips, and the PlayStation 5 chips, that AMD has sold for a short while under a different branding, as a CPU + MB for embedded systems.

**Comment by stefanstgs** [comment]
"https://web.archive.org/web/20220125033821/http://forum.notebookreview.com/threads/razer-blade-15-2021-rtx-3000.835061/page-20"
- With first link I want to refer to "MagillaGorilla" guide. He explains how to unlock bios on 10th gen intels ***and above. 

https://www.reddit.com/r/FlowZ13/comments/vplgfg/tutorial_compeletely_get_rid_of_the_dumb_cpu/
- This link goes more in-depth about how to use RU.exe to modify bios live *****. As I go through step above, to make bios writeable, I also usually disable OVERClOCKING_LOCK, which makes undervolting in windows possible on these machines.

https://web.archive.org/web/20220125202533/http://forum.notebookreview.com/threads/take-full-control-of-your-razer-blade-w-no-pl-throttling.827919/
https://web.archive.org/web/20220125095038/http://forum.notebookreview.com/threads/razer-bios-mods-potential-to-unlock-all-hidden-options.830993/
- This may be out of topic, but he also goes and talk about how to use the software to *edit bios. I usually run through these pages before I go and modify stuff. I have a pack with all the required software too, not sure I can distribute it all here though.

With these tuts u can then modify bios variables/display hidden menus through software used in tuts and then write it back without encrypted part.

***I am not sure as I have not tested past 10th gen intel, though, I think people in the thread mention doing it and working.

**** I am not sure exactly how to reach the proper variables which are editable with ru.efi like BIOS_LOCK but I am pretty sure it should be on the pages I posted. Otherwise let me know. This is the part I did once and forgot about. There are different variables names for different manufacturers, you can find them around the internet. Please use this safely. It seems difficult to undo changes that brick a machine and there is need for dedicated hardware.

How I know that my bios for gpu and uefi is on same chip is bcs I tried flashing a different bios for GPU and bios stuff got overwritten. Not exactly sure how it works still, or neither I am sure it is true that they are both on same chip.

Razer Blade Pro 17” (2020) | RZ09-0329x - My laptop model 

it has 2080 super max Q gpu.

I also use linux on machine and could try stuff if needed.

Thanks!
